src/app/services/token.interceptor.ts

```diff
@@ -0,0 +1,43 @@
+import { Injectable, Injector } from '@angular/core';
+import {
+  HttpEvent, HttpInterceptor, HttpHandler, HttpRequest, HttpErrorResponse
+} from '@angular/common/http';
+import { Router } from '@angular/router';
+import { Observable, throwError } from 'rxjs';
+import { catchError } from 'rxjs/operators';
+import { AccountService } from './account.service';
+
+
+@Injectable()
+export class TokenInterceptor implements HttpInterceptor {
+  private accountService: AccountService = null as any;
+  constructor(private injector: Injector) {}
+  intercept(request: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
+    this.accountService = this.injector.get(AccountService);
+    const accessToken: string | null = this.accountService.getToken('accessToken');
+    const headers: { [key: string]: string } = {
+      'Authorization': `Bearer ${accessToken}`,
+      'Content-Type': 'application/json'
+    };
+    request = request.clone({ setHeaders: headers });
+    return next.handle(request);
+  }
+}
+
+@Injectable()
```

Comment:
move this interceptor to separate file, 2 interceptors in same file is not so cool

Reply:
Fixed in A3